packages/backend/src/services/SlackService/SlackService.ts

```diff
@@ -0,0 +1,513 @@
+const uuidRegex = new RegExp(uuid, 'g');
+const nanoid = '[\\w-]{21}';
+const nanoidRegex = new RegExp(nanoid);
+
+export enum LightdashPage {
+    DASHBOARD = 'dashboard',
+    CHART = 'chart',
+    EXPLORE = 'explore',
+}
+
+type SlackServiceDependencies = {
+    lightdashConfig: LightdashConfig;
+    dashboardModel: DashboardModel;
+    savedChartModel: SavedChartModel;
+    spaceModel: SpaceModel;
+    shareModel: ShareModel;
+    encryptionService: EncryptionService;
+};
+
+const notifySlackError = async (
+    error: unknown,
+    url: string,
+    client: any,
+    event: any,
+): Promise<void> => {
+    console.error(`Unable to unfurl url ${JSON.stringify(error)}`);
+
+    const unfurls = {
+        [url]: {
+            blocks: [
+                {
+                    type: 'section',
+                    text: {
+                        type: 'mrkdwn',
+                        text: `Unable to unfurl URL ${url}: ${error} `,
+                    },
+                },
+            ],
+        },
+    };
+    await client.chat
+        .unfurl({
+            ts: event.message_ts,
+            channel: event.channel,
+            unfurls,
+        })
+        .catch((er: any) =>
+            console.error(`Unable to unfurl url ${JSON.stringify(er)}`),
+        );
+};
+
+const uploadImage = async (
+    screenshot: Buffer,
+    client: any,
+    event: any,
+    context: any,
+): Promise<string> => {
+    const imageId = event.message_ts;
+
+    // channel: event.channel and thread_ts:event.message_ts
+    // uploads the image to the same thread, however, unfurl is called while the user is typing, not when the message is sent
+    // so it is possible that the thread doesn't exist yet. Plus it is noisy (you get a notification when the image is uploaded)
+    // Instead we upload the image to the user personal channel (@user), always there, not noisy.
+    const slackUserId = await getSlackUserId(context);
+    const fileUpload = await client.files.upload({
+        channels: slackUserId, // event.channel
+        token: context.userToken,
+        file: screenshot,
+        filename: `dashboard-screenshot-${imageId}.png`,
+        // thread_ts: event.message_ts, // Upload on thread
+    });
+    const publicImage = await client.files.sharedPublicURL({
+        file: fileUpload.file.id,
+        token: context.userToken,
+    });
+
+    const permalink = publicImage?.file?.permalink_public;
+    const permalinkParts = permalink.split('-');
+    const pubSecret = permalinkParts[permalinkParts.length - 1];
+    const imageUrl = `${publicImage?.file.url_private}?pub_secret=${pubSecret}`;
+    return imageUrl;
+};
+
+const saveScreenshot = async (
+    imageId: string,
+    url: string,
+    cookie: string,
+    lightdashPage: LightdashPage,
+): Promise<Buffer> => {
+    let browser;
+
+    try {
+        const browserWSEndpoint = `ws://${process.env.HEADLESS_BROWSER_HOST}:${process.env.HEADLESS_BROWSER_PORT}`;
```

Comment:
We have been keeping the env vars usage/parsing centralised in `packages/backend/src/config/parseConfig.ts`. Should be move all the env var usage in this PR in there as well?

Reply:
yes 👍